### PR TITLE
[JBEAP-16152] Test checks if the identity sets by HttpServletRequest.login() is propagated into the secured EJB.

### PR DESCRIPTION
**Pull request by dcihak**
[JIRA issue](https://issues.jboss.org/browse/JBEAP-16152)

Test configures Elytron to use Identity Propagation.
Test deploys application with the secured servlet and checks if the identity sets by
HttpServletRequest.login() is propagated into the secured EJB.

### CODE REVIEW

**Comment by panossot** [comment]
When testing the PR I get these two errors : 

[ERROR] IdentityPropagationServlet.java:[25,82] package org.jboss.as.test.integration.web.security.identity.propagation.deployment does not exist
[ERROR] IdentityPropagationServerSetupTask.java:[28,65] package org.jboss.as.test.integration.web.security.authentication does not exist

File: modules/src/main/java/org/jboss/additional/testsuite/jdkall/present/web/security/identity/propagation/deployment/IdentityPropagationServlet.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.additional.testsuite.jdkall.present.web.security.identity.propagation.deployment;
+
+import org.jboss.additional.testsuite.jdkall.present.web.security.identity.propagation.IdentityPropagationAuthenticationTestCase;
+import org.jboss.as.test.integration.web.security.identity.propagation.deployment.Hello;
```
Comment (panossot) [inline modules/src/main/java/org/jboss/additional/testsuite/jdkall/present/web/security/identity/propagation/deployment/IdentityPropagationServlet.java]: This should probably be org.jboss.additional.testsuite.jdkall.present.web.security.identity.propagation.deployment.Hello;?

File: modules/src/main/java/org/jboss/additional/testsuite/jdkall/present/web/security/identity/propagation/deployment/IdentityPropagationServerSetupTask.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.additional.testsuite.jdkall.present.web.security.identity.propagation.deployment;
+
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.clustering.controller.Operations;
+import org.jboss.as.controller.client.helpers.ClientConstants;
+import org.jboss.as.test.integration.security.common.CoreUtils;
+import org.jboss.as.test.integration.web.security.authentication.BasicAuthMechanismServerSetupTask;
```
Comment (panossot) [inline modules/src/main/java/org/jboss/additional/testsuite/jdkall/present/web/security/identity/propagation/deployment/IdentityPropagationServerSetupTask.java]: BasicAuthMechanismServerSetupTask should also be added in this testsuite as in the server testsuite it is not part of the shared test classes.

**Comment by dcihak** [comment]
Thanks for the review, I am not sure how to work with the resources in EAP AT, hope this will fix it.

**Comment by panossot** [comment]
Now it fails with this message : [ERROR] org.jboss.additional.testsuite.jdkall.present.web.security.identity.propagation.IdentityPropagationAuthenticationTestCase  Time elapsed: 10.247 s  <<< ERROR!
org.jboss.arquillian.container.spi.client.container.DeploymentException: Cannot deploy httpRequestLogin.war: {"WFLYCTL0062: Composite operation failed and was rolled back. Steps that failed:" => {"Operation step-1" => {"WFLYCTL0412: Required services that are not installed:" => ["org.wildfly.ejb3.application-security-domain.auth-test"],"WFLYCTL0180: Services with missing/unavailable dependencies" => ["jboss.deployment.unit.\"httpRequestLogin.war\".ejb3.security-domain is missing [org.wildfly.ejb3.application-security-domain.auth-test]","jboss.deployment.unit.\"httpRequestLogin.war\".component.Hello.CREATE is missing [org.wildfly.ejb3.application-security-domain.auth-test]"]}}}

**Comment by panossot** [comment]
About the resources the default resource path is at the last level (test-configurations)/src/test/resources.
You can either distribute them which is preferable, as you have done in this PR, or add them there physically.

**Comment by panossot** [comment]
The problem is with the comment at the users and roles files.
You should add them manually at the resources,
here :  /test-configurations/src/test/resources/org/jboss/additional/testsuite/jdkall/present/web/security/identity/propagation , without the @EapAdditionalTestsuite annotation
Then I will merge it and I will move it to another test-configuration dir and it will work.

**Comment by dcihak** [comment]
Thank you for the investigation, PR updated.

**Comment by panossot** [comment]
I am merging. I will move it to another test-configuration dir and it will work.

**Pull request merged**